pylint/test/functional/missing_parentheses_for_call_in_test.py

```diff
@@ -0,0 +1,80 @@
+"""Verify if call to function or method inside tests are missing parentheses."""
+# pylint: disable=using-constant-test, missing-docstring, useless-object-inheritance
+# pylint: disable=invalid-name, expression-not-assigned
+
+import collections
+
+
+def bool_function():
+    return True
+
+def nonbool_function():
+    return 42
+
+
+class Class(object):
+
+    @staticmethod
+    def bool_method():
+        return False
+
+    @staticmethod
+    def nonbool_method():
+        return 42
+
+
+if collections:
+    pass
+
+if bool_function: # [missing-parentheses-for-call-in-test]
+    pass
+
+if not bool_function():
+    pass
+
+if nonbool_function:  # [missing-parentheses-for-call-in-test]
+    pass
+
+if nonbool_function() != 42:
+    pass
+
+instance = Class()
+
+if instance.bool_method:  # [missing-parentheses-for-call-in-test]
+    pass
+
+if not instance.bool_method():
+    pass
+
+if not instance.nonbool_method:
+    pass
+elif instance.bool_method:  # [missing-parentheses-for-call-in-test]
+    pass
+
+bool_lambda = lambda: True
+
+if bool_lambda:  # [missing-parentheses-for-call-in-test]
+    pass
+
+if not bool_lambda():
+    pass
+
+nonbool_lambda = lambda: 42
```

Comment:
What happens if we don't return a `Const` node? Say a `list` or something else? Would this check trigger? From line 1066 in the check file it seems not.